### PR TITLE
LmP staging bitbake class

### PR DESCRIPTION
**Pull request by quaresmajose**
This class will implement some pending patches that we have
and some workarounds needed in LmP.

For now there is only fixes for:
- linux kernel modules dependencies and inheritance
- network during compile for go-mod.bbclass

### CODE REVIEW

**Review by MrCry0 — APPROVED** [review]
I like this :)
LGTM

**Review by angolini — APPROVED** [review]

**Comment by angolini** [comment]
Could you please add a description to the PR?

So, it will be easier to know how to use the class in the future.

**Comment by quaresmajose** [comment]
Sure, I added a little description

File: meta-lmp-base/classes/lmp-staging.bbclass [hunk 1]
```diff
@@ -12,6 +12,9 @@ python __anonymous() {
         d.appendVar('DEPENDS', ' virtual/kernel')
         if 'modsign' in d.getVar('DISTRO_FEATURES'):
             d.setVar('INHERIT_KERNEL_MODSIGN', 'kernel-modsign')
+
+    if bb.data.inherits_class('go-mod', d):
+            d.appendVarFlag('do_compile', 'network', '1')
```
Comment (ricardosalveti) [inline meta-lmp-base/classes/lmp-staging.bbclass:17]: Wonder if the lack of proxy will cause the older issue to happen again, will do a local test.

Comment (quaresmajose) [reply meta-lmp-base/classes/lmp-staging.bbclass:17]: the use of an invalid proxy cause a build fail https://github.com/foundriesio/meta-lmp/pull/824#issuecomment-1263801790 but I don't remember to did the test without proxy. so I will do that as well

Comment (quaresmajose) [reply meta-lmp-base/classes/lmp-staging.bbclass:17]: Tested without proxy https://github.com/foundriesio/meta-lmp/pull/824#issuecomment-1276476242 and works in the same way.

Comment (ricardosalveti) [reply meta-lmp-base/classes/lmp-staging.bbclass:17]: Cool, thanks!

Just noticed there is 4 extra spaces here, can you fix to 4 to be aligned with the lines above it?

Comment (quaresmajose) [reply meta-lmp-base/classes/lmp-staging.bbclass]: Fixed, I don't even know how python didn't complain about the indentation!

Comment (ricardosalveti) [reply meta-lmp-base/classes/lmp-staging.bbclass]: I guess it would complain if it was a tab instead.

**Review by ricardosalveti — APPROVED** [review]
LGTM

**Pull request merged**